python/desc/imsim/camera_readout.py

```diff
@@ -273,12 +291,19 @@ def get_amplifier_hdu(self, amp_name):
             Image HDU with the pixel data and header keywords
             appropriate for the requested sensor segment.
         """
-        hdu = fits.ImageHDU(data=self.amp_images[amp_name].getArray(),
-                            do_not_scale_image_data=True)
+        hdu = fits.ImageHDU(data=self.amp_images[amp_name].getArray().astype(np.int32))
```

Comment:
I see you lost: "do_not_scale_image_data=True"

Is that because you are explicitly setting the type here with as type so you don't have to worry?

Reply:
that option resulted in the files being written as floats, which isn't consistent with the real data and which does not compress as well as ints.